### PR TITLE
Fixes #787

### LINKED ISSUE #787: comments on Q&A slightly wonky
### What happened just before the problem occurred

an accepted answer in the new Q&A system shows a count of 1 comments, but there is no comment. 

screen shot 2016-09-03 at 4 58 55 pm

### Relevant URLs

https://publiclab.org/questions/nshapiro/09-02-2016/question-what-is-the-open-source-community-s-take-the-mit-license#a33

### PublicLab.org username

 liz 

### Browser, version, and operating system

### PR DESCRIPTION
**Pull request by ananyo2012**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] All tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [x] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by ananyo2012** [comment]
@jywarren Please review

File: app/views/comment_mailer/notify.html.erb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Hi! There's been a response to a discussion you're involved in. Do NOT reply to this email; click this link to respond:
 
 <% if @comment.parent.has_power_tag('question') %>
-<p>https://publiclab.org<%= @comment.parent.path(:question) %>#answer-comment-<%= @comment.cid %></p>
+<p>https://publiclab.org<%= @comment.parent.path(:question) %>#answer-<%= @comment.aid %>-comment-<%= @comment.cid %></p>
```
Comment (ananyo2012) [inline app/views/comment_mailer/notify.html.erb]: Corrected the urls for comments in email notifications

**Comment by jywarren** [comment]
Is this rebased on top of today's changes? Also, feel free to close and reopen to trigger retesting if it gets stuck on the jasmine tests.

**Comment by ananyo2012** [comment]
Ok i see one failing test. Sorry i didn't check. Yes this is rebased. I will sending in the revisions soon.

**Comment by jywarren** [comment]
re-running this one to try to get it to run the Jasmine tests properly.

**Comment by jywarren** [comment]
We're getting:

```
bower flot#~0.7                          ECMDERR Failed to execute "git ls-remote --tags --heads https://github.com/flot/flot.git", exit code of #128 remote: Internal Server Error. remote:  fatal: unable to access 'https://github.com/flot/flot.git/': The requested URL returned error: 500
Additional error details:
remote: Internal Server Error.
remote: 
fatal: unable to access 'https://github.com/flot/flot.git/': The requested URL returned error: 500
```

That's odd, as that URL resolves fine for me. I'll try running one more time.

**Comment by jywarren** [comment]
Hmm, @icarito -- maybe that version of phantomjs is no longer hosted? 

```

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 32 22.3M   32 7428k    0     0  19692      0  0:19:49  0:06:26  0:13:23     0curl: (56) SSL read: error:00000000:lib(0):func(0):reason(0), errno 104
rake aborted!
Failed to load phantomjs! :(
You need to have cURL or wget installed on your system.
If you have, the source of phantomjs might be unavailable: https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2
```

But it looks like it is -- maybe it's just timing out or something? It seems to be downloading something. And https://bitbucket.org/ariya/phantomjs/downloads/ includes the exact file we want. Ah, there's a `curl` error: 

`curl: (56) SSL read: error:00000000:lib(0):func(0):reason(0), errno 104`

It looks like maybe we just need to cache that binary after all to make JS tests a bit more stable.

**Comment by jywarren** [comment]
Could we use Travis caching for the directory mentioned in the message: `Phantomjs does not appear to be installed in /root/.phantomjs/2.1.1/x86_64-linux/bin/phantomjs, installing!`

phew, it finally passed, though.

**Comment by ananyo2012** [comment]
Yes I already mentioned previously we should cache the phantomjs
installation to ensure that js tests don't fail due to a installation
problem.
ᐧ

**Comment by icarito** [comment]
Indeed travis pulls this file so many times it causes the server to throttle and reject clients.
I guess this is a shortfall of centralized public testing.

We can cache ourselves, but i'd like a robust method. I'll try it.

**Comment by icarito** [comment]
I tried this approach it seems to work reliably. Since we build the whole container for testing, we should figure out another way to cache. Possibly we can try to cache `/var/docker` to cache build steps and save time.

**Pull request merged**